pkg/async/pool/timeout_test.go

```diff
@@ -14,32 +14,38 @@ func TestTimeout(t *testing.T) {
 	ctx, cancel := context.WithTimeout(context.Background(), 5*time.Second)
 	defer cancel()
 
-	p := pool.New(ctx,
-		pool.ConstantSize(1),
-	)
+	p := pool.New(ctx, pool.ConstantSize(1))
 
 	startedAt := time.Now()
 
-	scheduler := pool.WithTimeout(5*time.Millisecond, p)
+	timeout := 5 * time.Millisecond
+	scheduler := pool.WithTimeout(timeout, p)
 
 	scheduler, wait := pool.WithWait(scheduler)
 
 	var err error
+	// The signal to start first scheduled task.
+	startFirst := make(chan bool)
+	defer close(startFirst)
 
 	scheduler.Schedule(ctx, async.Func(func(ctx context.Context) error {
 		if ctx.Err() != nil {
 			return ctx.Err()
 		}
-		time.Sleep(10 * time.Millisecond)
+		<-startFirst
+		// Ensure the next task times out.
+		time.Sleep(2 * timeout)
```

Comment:
Great idea on ensuring this timeout!

Reply:
You complementing the variable? I'm actually generally terribly lazy when it comes to tests, but this helped the readability here. :)